src/monster.h

```diff
@@ -414,7 +414,7 @@ class monster : public Creature
         int wandf;           // Urge to wander - Increased by sound, decrements each move
         std::vector<item> inv; // Inventory
         player *dragged_foe; // player being dragged by the monster
-
+        itype_id tied_item; // item used to tie the monster
```

Comment:
You might as well store the whole item here. You current change allows to tie something with a *damaged* rope item, but untieing it will yield a normal (undamaged) item.

Reply:
This is possible even without this change. There is even an issue for this, I can't find it's number though.

Reply:
Yeah it currently takes any short rope in your inventory, then spawns a fresh one when untied.
But I can store the actual item.
Question though - is it easier/better to store a pointer to the item itself in monster.h  ? or just item.

Reply:
Store the item as `cata::optional<item>`. This allows to store nothing (when not tied).